### PR TITLE
docs: better compatibility matrix

### LINKED ISSUE #5841: Documentation Compatibility Matrix
### Description

**How can the docs be improved?**
Improve the compatibility matrix, by changing the Semantic Version of Karpenter in the matrix.
Instead of using Major.Minor.Patch+, replace with >= Major.Minor.Patch

Example:
Karpenter 0.28.x+ is matched with Kubernetes 1.26 and 1.27 in the current matrix, users confuse that they are unable to use Karpenter 0.33 with Kubernetes 1.26 and 1.27 and they can only run patch releases of 0.28 on Kubernetes 1.26 and 1.27
Changing the Karpenter semantic version comparison to ">= 0.28" is more clear.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by erezzarum**
Fixes #5841  

**Description**
Change semantic version comparison for Karpenter/Kubernetes in the Compatibility Matrix.

**How was this change tested?**

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [x] Yes, issue opened: #5841  
- [] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/docs/upgrading/compatibility.md [hunk 1]
```diff
@@ -15,9 +15,9 @@ Before you begin upgrading Karpenter, consider Karpenter compatibility issues re
 
 [comment]: <> (the content below is generated from hack/docs/compataiblitymetrix_gen_docs.go)
 
-| KUBERNETES |  1.23   |  1.24   |  1.25   |  1.26   |  1.27   |  1.28   |  1.29   |
-|------------|---------|---------|---------|---------|---------|---------|---------|
-| karpenter  | 0.21.x+ | 0.21.x+ | 0.25.x+ | 0.28.x+ | 0.28.x+ | 0.31.x+ | 0.34.0+ |
+| KUBERNETES | 1.23     | 1.24     | 1.25     | 1.26     | 1.27     | 1.28     | 1.29       |
```
Comment (jonathan-innis) [inline website/content/en/docs/upgrading/compatibility.md:18]: This data is generated. If this is a change that we want to make, I think we need to make a change to https://github.com/aws/karpenter-provider-aws/tree/main/tools/kompat to generate different data

Comment (erezzarum) [reply website/content/en/docs/upgrading/compatibility.md]: reverted back the doc and committed a fix to kompat

File: website/content/en/preview/upgrading/compatibility.md [hunk 2]
```diff
@@ -15,9 +15,9 @@ Before you begin upgrading Karpenter, consider Karpenter compatibility issues re
 
 [comment]: <> (the content below is generated from hack/docs/compataiblitymetrix_gen_docs.go)
 
-| KUBERNETES |  1.23   |  1.24   |  1.25   |  1.26   |  1.27   |  1.28   |  1.29   |
-|------------|---------|---------|---------|---------|---------|---------|---------|
-| karpenter  | 0.21.x+ | 0.21.x+ | 0.25.x+ | 0.28.x+ | 0.28.x+ | 0.31.x+ | 0.34.0+ |
+| KUBERNETES |   1.23   |   1.24   |   1.25   |   1.26   |   1.27   |   1.28   |    1.29    |
```
Comment (engedaam) [inline website/content/en/preview/upgrading/compatibility.md:18]: Can you copy these changes into all the version of the docs?

Comment (erezzarum) [reply website/content/en/preview/upgrading/compatibility.md:18]: Committed

**Review by engedaam — APPROVED** [review]
LGTM 🚀

**Pull request merged**